XenAdmin/Dialogs/GraphDetailsDialog.cs

```diff
@@ -190,8 +190,11 @@ private void EnableDatasource()
             if (graphList.XenObject?.Connection == null || dataGridView.SelectedRows.Count != 1)
                 return;
 
-            var row = dataGridView.SelectedRows[0] as DataSourceGridViewRow;
-            var dataSource = row?.Dsi.DataSource;
+            if (!(dataGridView.SelectedRows[0] is DataSourceGridViewRow row))

```

Comment:
not sure why there is also a type check, since the `Count` is checked right before (line 190).

I wasn't making any functionality changes so I didn't make changes but if this is useless we can remove it?

Maybe @kc284 can advise